### ISSUE TITLE
The first method (WidgetTester.pump) had not yet finished executing at the time that the second

### ISSUE THREAD

**Issue by yahyakara** [jonsamwell/flutter_gherkin#198]
Hi everyone, I use the integration test rc 9 package for our test, If the first scenario fails I gets an error on the second test starts. I use the reset hook as below, 

`@override
  Future<void> onAfterScenario(
    TestConfiguration config,
    String scenario,
    Iterable<Tag> tags,
  ) async {
    await (await SharedPreferences.getInstance()).clear();
    await getIt.reset(dispose: false);
  }`

 I use the getIt.reset method for the reseting states of app if I don't use it I get some error about app states. For example I can't add product to cart.
 But with  getIt.reset(dispose: false); I gets error as below and test crashes on first method of second test, 

``` I/flutter (22362): ┌───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
I/flutter (22362): │ 00:23:34.378 (+0:00:18.188623)
I/flutter (22362): ├┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄
I/flutter (22362): │ 💡 AppName UAT - com.package..uat - 0.1.0 - (1207)
I/flutter (22362): └───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
D/Sentry  (22362): UncaughtExceptionHandlerIntegration removed.
D/sentry-native(22362): shutting down backend
D/Sentry  (22362): NdkIntegration removed.
D/Sentry  (22362): EnvelopeFileObserverIntegration removed.
D/Sentry  (22362): AppLifecycleIntegration removed.
D/Sentry  (22362): ActivityLifecycleIntegration removed.
D/Sentry  (22362): AppComponentsBreadcrumbsIntegration removed.
D/Sentry  (22362): SystemEventsBreadcrumbsIntegration remove.
D/Sentry  (22362): TempSensorBreadcrumbsIntegration removed.
D/Sentry  (22362): Envelope sent successfully.
D/Sentry  (22362): Discarding envelope from cache: /data/user/0/com.package..uat/cache/sentry/9d3e7241-8c7f-40d3-926c-b1017fe91f61.envelope
D/Sentry  (22362): Envelope flushed
I/System.out(22362): (HTTPLog)-Static: isSBSettingEnabled false
I/System.out(22362): (HTTPLog)-Static: isSBSettingEnabled false
D/Sentry  (22362): Serializing object:
{"exception":{"values":[{"mechanism":{"handled":true,"type":"runZonedGuarded"},"stacktrace":{"frames":[{"abs_path":"<asynchronous
suspension>"},{"abs_path":"<asynchronous
suspension>"},{"package":"name","abs_path":"package/main.dart","colno":3,"filename":"main.dart","function":"_runApp","in_app":true,"lineno":41,"unknown":{"pa
ckage":"name"}},{"abs_path":"<asynchronous
suspension>"},{"package":"name","abs_path":"package/core/injection_container.dart","colno":23,"filename":"injection_container.dart","function":"initDependenc
ies","in_app":true,"lineno":220,"unknown":{"package":"name"}},{"package":"name","abs_path":"package/core/network/graphql/shopify_graphql_client.dart","colno
":11,"filename":"shopify_graphql_client.dart","function":"ShopifyGraphQlClient.init","in_app":true,"lineno":25,"unknown":{"package":"name"}},{"package":"name","ab
s_path":"package/core/network/graphql/shopify_graphql_client.dart","colno":34,"filename":"shopify_graphql_client.dart","function":"ShopifyGraphQlClient._refre
shClientConfig","in_app":true,"lineno":38,"unknown":{"package":"name"}},{"package":"get_it","abs_path":"package:get_it/get_it_impl.dart","colno":12,"filename":"get
_it_impl.dart","function":"_GetItImplementation.call","in_app":true,"lineno":430,"unknown":{"package":"get_it"}},{"package":"get_it","abs_path":"package:get_it/get_
it_impl.dart","colno":34,"filename":"get_it_impl.dart","function":"_GetItImplementation.get","in_app":true,"lineno":409,"unknown":{"package":"get_it"}},{"package":"
get_it","abs_path":"package:get_it/get_it_impl.dart","colno":37,"filename":"get_it_impl.dart","function":"_ServiceFactory.getObject","in_app":true,"lineno":154,"unk
nown":{"package":"get_it"}},{"package":"name","abs_path":"package/core/injection_container.dart","colno":89,"filename":"injection_container.dart","function":
"_registerUseCases.<fn>","in_app":true,"lineno":412,"unknown":{"package":"name"}},{"package":"get_it","abs_path":"package:get_it/get_it_impl.dart","colno":12,"file
name":"get_it_impl.dart","function":"_GetItImplementation.call","in_app":true,"lineno":430,"unknown":{"package":"get_it"}},{"package":"get_it","abs_path":"package:g
et_it/get_it_impl.dart","colno":34,"filename":"get_it_impl.dart","function":"_GetItImplementation.get","in_app":true,"lineno":409,"unknown":{"package":"get_it"}},{"
package":"get_it","abs_path":"package:get_it/get_it_impl.dart","colno":41,"filename":"get_it_impl.dart","function":"_ServiceFactory.getObject","in_app":true,"lineno
":160,"unknown":{"package":"get_it"}},{"package":"name","abs_path":"package/core/injection_container.dart","colno":83,"filename":"injection_container.dart","
function":"_registerRepositories.<fn>","in_app":true,"lineno":299,"unknown":{"package":"name"}},{"package":"get_it","abs_path":"package:get_it/get_it_impl.dart","c
olno":12,"filename":"get_it_impl.dart","function":"_GetItImplementation.call","in_app":true,"lineno":430,"unknown":{"package":"get_it"}},{"package":"get_it","abs_pa
th":"package:get_it/get_it_impl.dart","colno":29,"filename":"get_it_impl.dart","function":"_GetItImplementation.get","in_app":true,"lineno":393,"unknown":{"package"
:"get_it"}},{"package":"get_it","abs_path":"package:get_it/get_it_impl.dart","colno":7,"filename":"get_it_impl.dart","function":"_GetItImplementation._findFactoryBy
NameAndType","in_app":true,"lineno":372,"unknown":{"package":"get_it"}},{"abs_path":"dart:core-patch/errors_patch.dart","colno":5,"filename":"errors_patch.dart","fu
nction":"_AssertionError._throwNew","in_app":false,"lineno":36},{"abs_path":"dart:core-patch/errors_patch.dart","colno":61,"filename":"errors_patch.dart","function"
:"_AssertionError._doThrowNew","in_app":false,"lineno":47}]},"type":"_AssertionError","value":"'package:get_it/get_it_impl.dart': Failed assertion: line 372 pos 7:
'instanceFactory != null': Object/factory with  type CountryLocalDataSource is not registered inside GetIt. \n(Did you accidentally do GetIt sl=GetIt.instance();
instead of GetIt sl=GetIt.instance;\nDid you forget to
D/Braze v17.0.0 .bo.app.q5(22362): Request(id = 8c35f967ccb5d420) Executed in [383ms] [POST:https://sdk.iad-01.braze.com/api/v3/data]
D/Braze v17.0.0 .bo.app.d3(22362): Result(id = 8c35f967ccb5d420) 
D/Braze v17.0.0 .bo.app.d3(22362): {}
D/Braze v17.0.0 .bo.app.b6(22362): Triggered actions Json array was null. Not de-serializing triggered actions.
D/Braze v17.0.0 .bo.app.b6(22362): Templated message Json was null. Not de-serializing templated message.
D/Braze v17.0.0 .h0(22362): DataSyncRequest executed successfully.
V/Braze v17.0.0 .bo.app.t(22362): Processing server response payload for user with id: 
D/Braze v17.0.0 .bo.app.b1(22362): bo.app.m4 fired: bo.app.m4@f81630d
D/Braze v17.0.0 .bo.app.b1(22362): Triggering bo.app.m4 on 1 subscribers.
D/Braze v17.0.0 .bo.app.b1(22362): bo.app.q0 fired: bo.app.q0@f5352c2
D/Braze v17.0.0 .bo.app.b1(22362): Triggering bo.app.q0 on 1 subscribers.
V/Braze v17.0.0 .bo.app.a(22362): Notifying confirmAndUnlock listeners for cache: bo.app.j0@87582fa
V/Braze v17.0.0 .bo.app.a(22362): Notifying confirmAndUnlock listeners for cache: bo.app.i6@f4d20ab
D/Sentry  (22362): Envelope sent successfully.
D/Sentry  (22362): Envelope flushed
D/Sentry  (22362): Marking envelope submission result: true
D/Sentry  (22362): File '/data/user/0/com.package..uat/cache/sentry/outbox/d0e9a537-0435-4a95-901a-e93837907a29' is done.
D/Sentry  (22362): Finished processing cached files from /data/user/0/com.package..uat/cache/sentry/outbox
I/System.out(22362): (HTTPLog)-Static: isSBSettingEnabled false
I/System.out(22362): (HTTPLog)-Static: isSBSettingEnabled false
D/Sentry  (22362): Envelope sent successfully.
D/Sentry  (22362): Envelope flushed
D/Sentry  (22362): Marking envelope submission result: true
D/Sentry  (22362): Deleted file /data/user/0/com.package..uat/cache/sentry/9a40f1d9-e6a2-4606-b335-7f66398090b9.envelope.
D/Sentry  (22362): Finished processing cached files from /data/user/0/com.package..uat/cache/sentry
D/Sentry  (22362): Started processing cached files from /data/user/0/com.package..uat/cache/sentry/outbox
D/Sentry  (22362): Processing dir. /data/user/0/com.package..uat/cache/sentry/outbox
D/Sentry  (22362): Processing 2 items from cache dir /data/user/0/com.package..uat/cache/sentry/outbox
D/Sentry  (22362): Processing file: /data/user/0/com.package..uat/cache/sentry/outbox/d0a09a9e-8ca7-4f0e-af98-d6f369f64c1a
D/Sentry  (22362): Processing Envelope with 1 item(s)
D/Sentry  (22362): Item 1 is being captured.
I/Sentry  (22362): Closing SentryClient.
D/Sentry  (22362): Shutting down
E/Sentry  (22362): Exception while awaiting on lock.
E/Sentry  (22362): java.lang.InterruptedException
E/Sentry  (22362):      at java.util.concurrent.locks.AbstractQueuedSynchronizer.doAcquireSharedNanos(AbstractQueuedSynchronizer.java:1065)
E/Sentry  (22362):      at java.util.concurrent.locks.AbstractQueuedSynchronizer.tryAcquireSharedNanos(AbstractQueuedSynchronizer.java:1358)
E/Sentry  (22362):      at java.util.concurrent.CountDownLatch.await(CountDownLatch.java:278)
E/Sentry  (22362):      at io.sentry.DirectoryProcessor$SendCachedEnvelopeHint.waitFlush(DirectoryProcessor.java:105)
E/Sentry  (22362):      at io.sentry.OutboxSender.waitFlush(OutboxSender.java:245)
E/Sentry  (22362):      at io.sentry.OutboxSender.processEnvelope(OutboxSender.java:137)
E/Sentry  (22362):      at io.sentry.OutboxSender.processFile(OutboxSender.java:70)
E/Sentry  (22362):      at io.sentry.DirectoryProcessor.processDirectory(DirectoryProcessor.java:66)
E/Sentry  (22362):      at io.sentry.OutboxSender.processDirectory(OutboxSender.java:29)
E/Sentry  (22362):      at
io.sentry.SendCachedEnvelopeFireAndForgetIntegration$SendFireAndForgetFactory$-CC.lambda$processDir$0(SendCachedEnvelopeFireAndForgetIntegration.java:42)
E/Sentry  (22362):      at io.sentry.-$$Lambda$SendCachedEnvelopeFireAndForgetIntegration$SendFireAndForgetFactory$fw_M6vxYehvhSeLyExDmpVukEZU.send(Unknown Source:8)
E/Sentry  (22362):      at io.sentry.SendCachedEnvelopeFireAndForgetIntegration.lambda$register$0(SendCachedEnvelopeFireAndForgetIntegration.java:79)
E/Sentry  (22362):      at io.sentry.-$$Lambda$SendCachedEnvelopeFireAndForgetIntegration$-yc-vZFUWWVM7GTAiUeYZz2iHz4.run(Unknown Source:4)
E/Sentry  (22362):      at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:462)
E/Sentry  (22362):      at java.util.concurrent.FutureTask.run(FutureTask.java:266)
E/Sentry  (22362):      at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
E/Sentry  (22362):      at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
E/Sentry  (22362):      at java.lang.Thread.run(Thread.java:923)
W/Sentry  (22362): Timed out waiting for event id submission: d01fecaf6ee94faeaff23d792025e091
D/Sentry  (22362): File '/data/user/0/com.package..uat/cache/sentry/outbox/d0a09a9e-8ca7-4f0e-af98-d6f369f64c1a' is done.
D/Sentry  (22362): Processing file: /data/user/0/com.package..uat/cache/sentry/outbox/66aa7008-873c-426f-b742-cf4801538f50
I/flutter (22362): PASSED: Scenario Anonymous user should redirect to login screen while enter the basket if he added product # :0
D/Sentry  (22362): Processing Envelope with 1 item(s)
I/Sentry  (22362): Initializing SDK with DSN: 'https://80b9c58f7def462c8e1f6b03b3723292@o1018340.ingest.sentry.io/5984094'
I/Sentry  (22362): GlobalHubMode: 'true'
D/Sentry  (22362): UncaughtExceptionHandlerIntegration enabled: true
D/Sentry  (22362): default UncaughtExceptionHandler class='com.google.firebase.crashlytics.internal.common.CrashlyticsUncaughtExceptionHandler'
D/Sentry  (22362): UncaughtExceptionHandlerIntegration installed.
D/Sentry  (22362): ShutdownHookIntegration installed.
D/Sentry  (22362): Capturing event: 1ebd13176dc740139373dc29b84f7b5f
D/Sentry  (22362): Event was cached so not applying scope: 1ebd13176dc740139373dc29b84f7b5f
D/Sentry  (22362): Event was cached so not applying data relevant to the current app execution/version: 1ebd13176dc740139373dc29b84f7b5f
D/Sentry  (22362): Event was cached so not applying data relevant to the current app execution/version: 1ebd13176dc740139373dc29b84f7b5f
D/Sentry  (22362): SendCachedEventFireAndForgetIntegration installed.
D/Sentry  (22362): NdkIntegration enabled: true
D/Sentry  (22362): Started processing cached files from /data/user/0/com.package..uat/cache/sentry
I/sentry-native(22362): sentry_close() called, but options was empty
I/sentry-native(22362): using database path "/data/data/com.package..uat/cache/sentry/.sentry-native"
D/Sentry  (22362): Processing dir. /data/user/0/com.package..uat/cache/sentry
D/sentry-native(22362): starting backend
D/Sentry  (22362): NdkIntegration installed.
D/Sentry  (22362): Registering EnvelopeFileObserverIntegration for path: /data/user/0/com.package..uat/cache/sentry/outbox
D/Sentry  (22362): EnvelopeFileObserverIntegration installed.
D/Sentry  (22362): Processing 1 items from cache dir /data/user/0/com.package..uat/cache/sentry
D/Sentry  (22362): SendCachedEventFireAndForgetIntegration installed.
D/Sentry  (22362): AnrIntegration enabled: false
D/Sentry  (22362): enableSessionTracking enabled: true
D/Sentry  (22362): File /data/user/0/com.package..uat/cache/sentry/outbox is not a File.
D/Sentry  (22362): enableAppLifecycleBreadcrumbs enabled: true
D/Sentry  (22362): SU isn't found on this Device.
D/Sentry  (22362): File /data/user/0/com.package..uat/cache/sentry/.sentry-native is not a File.
D/Sentry  (22362): Processing file: /data/user/0/com.package..uat/cache/sentry/session.json
D/Sentry  (22362): File '/data/user/0/com.package..uat/cache/sentry/session.json' doesn't match extension expected.
D/Sentry  (22362): Processing file: /data/user/0/com.package..uat/cache/sentry/9344092b-1944-42ba-82f9-414ca803c938.envelope
D/Sentry  (22362): AppLifecycleIntegration installed.
D/Sentry  (22362): ActivityLifecycleIntegration enabled: true
D/Sentry  (22362): ActivityLifecycleIntegration installed.
D/Sentry  (22362): Captured Envelope is already cached
D/Sentry  (22362): AppComponentsBreadcrumbsIntegration enabled: true
D/Sentry  (22362): AppComponentsBreadcrumbsIntegration installed.
D/Sentry  (22362): SystemEventsBreadcrumbsIntegration enabled: true
W/Sentry  (22362): Current session is not ended, we'd need to end it.
D/Sentry  (22362): SystemEventsBreadcrumbsIntegration installed.
D/Sentry  (22362): enableSystemEventsBreadcrumbs enabled: true
I/Sentry  (22362): TYPE_AMBIENT_TEMPERATURE is not available.
D/Sentry  (22362): enableSystemEventBreadcrumbs enabled: true
D/Sentry  (22362): Serializing object:
{"sid":"948ea142-0541-4b47-91aa-83e5508ba6d4","did":"f13dc5cc-5888-4c0f-99f3-0ce40d2477c2","started":"2022-01-25T21:23:34.285Z","status":"exited","seq":164314581641
9,"duration":2.134,"timestamp":"2022-01-25T21:23:36.419Z","attrs":{"release":"com.package..uat@0.1.0+1207","environment":".uat"}}
D/Sentry  (22362): Serializing object:
{"sid":"0a783cdf-c30c-4c00-bee4-3b88b9fd9004","did":"f13dc5cc-5888-4c0f-99f3-0ce40d2477c2","init":true,"started":"2022-01-25T21:23:36.415Z","status":"ok","timestamp
":"2022-01-25T21:23:36.415Z","attrs":{"release":"com.package..uat@0.1.0+1207","environment":".uat"}}
D/Sentry  (22362): Serializing object:
{"sid":"0a783cdf-c30c-4c00-bee4-3b88b9fd9004","did":"f13dc5cc-5888-4c0f-99f3-0ce40d2477c2","init":true,"started":"2022-01-25T21:23:36.415Z","status":"ok","timestamp
":"2022-01-25T21:23:36.415Z","attrs":{"release":"com.package..uat@0.1.0+1207","environment":".uat"}}
I/Sentry  (22362): Not possible to read external files directory
D/Sentry  (22362): Adding Envelope to offline storage: /data/user/0/com.package..uat/cache/sentry/c23b7f17-7e92-49ff-880c-39feefb9cb28.envelope
D/Sentry  (22362): Captured Envelope is already cached
D/Sentry  (22362): Item 1 is being captured.
E/Sentry  (22362): Exception while awaiting on lock.
E/Sentry  (22362): java.lang.InterruptedException
E/Sentry  (22362):      at java.util.concurrent.locks.AbstractQueuedSynchronizer.tryAcquireSharedNanos(AbstractQueuedSynchronizer.java:1356)
E/Sentry  (22362):      at java.util.concurrent.CountDownLatch.await(CountDownLatch.java:278)
E/Sentry  (22362):      at io.sentry.DirectoryProcessor$SendCachedEnvelopeHint.waitFlush(DirectoryProcessor.java:105)
E/Sentry  (22362):      at io.sentry.OutboxSender.waitFlush(OutboxSender.java:245)
E/Sentry  (22362):      at io.sentry.OutboxSender.processEnvelope(OutboxSender.java:137)
E/Sentry  (22362):      at io.sentry.OutboxSender.processFile(OutboxSender.java:70)
E/Sentry  (22362):      at io.sentry.DirectoryProcessor.processDirectory(DirectoryProcessor.java:66)
E/Sentry  (22362):      at io.sentry.OutboxSender.processDirectory(OutboxSender.java:29)
E/Sentry  (22362):      at
io.sentry.SendCachedEnvelopeFireAndForgetIntegration$SendFireAndForgetFactory$-CC.lambda$processDir$0(SendCachedEnvelopeFireAndForgetIntegration.java:42)
E/Sentry  (22362):      at io.sentry.-$$Lambda$SendCachedEnvelopeFireAndForgetIntegration$SendFireAndForgetFactory$fw_M6vxYehvhSeLyExDmpVukEZU.send(Unknown Source:8)
E/Sentry  (22362):      at io.sentry.SendCachedEnvelopeFireAndForgetIntegration.lambda$register$0(SendCachedEnvelopeFireAndForgetIntegration.java:79)
E/Sentry  (22362):      at io.sentry.-$$Lambda$SendCachedEnvelopeFireAndForgetIntegration$-yc-vZFUWWVM7GTAiUeYZz2iHz4.run(Unknown Source:4)
E/Sentry  (22362):      at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:462)
E/Sentry  (22362):      at java.util.concurrent.FutureTask.run(FutureTask.java:266)
E/Sentry  (22362):      at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
E/Sentry  (22362):      at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
E/Sentry  (22362):      at java.lang.Thread.run(Thread.java:923)
W/Sentry  (22362): Timed out waiting for event id submission: 1ebd13176dc740139373dc29b84f7b5f
D/Sentry  (22362): File '/data/user/0/com.package..uat/cache/sentry/outbox/66aa7008-873c-426f-b742-cf4801538f50' is done.
D/Sentry  (22362): Finished processing cached files from /data/user/0/com.package..uat/cache/sentry/outbox
I/System.out(22362): (HTTPLog)-Static: isSBSettingEnabled false
I/System.out(22362): (HTTPLog)-Static: isSBSettingEnabled false
I/flutter (22362): ══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞════════════════════════════════════════════════════
I/flutter (22362): The following assertion was thrown running a test:
I/flutter (22362): Guarded function conflict.
I/flutter (22362): You must use "await" with all Future-returning test APIs.
I/flutter (22362): The guarded method "pump" from class WidgetTester was called from
I/flutter (22362): file:///Users/yahya.kara/AndroidStudioProjects/dh-name-flutter/test_screens/utils/general_test_utils.dart
I/flutter (22362): on line 32.
**I/flutter (22362): Then, the "pumpAndSettle" method (also from class WidgetTester) was called from
I/flutter (22362): package:flutter_gherkin/src/flutter/runners/gherkin_integration_test_runner.dart on line 226.
I/flutter (22362): The first method (WidgetTester.pump) had not yet finished executing at the time that the second
I/flutter (22362): method (WidgetTester.pumpAndSettle) was called. Since both are guarded, and the second was not a
I/flutter (22362): nested call inside the first, the first must complete its execution before the second can be called.
I/flutter (22362): Typically, this is achieved by putting an "await" statement in front of the call to the first.**
I/flutter (22362): 
I/flutter (22362): When the first method (WidgetTester.pump) was called, this was the stack:
I/flutter (22362): #0      TestAsyncUtils.guard (package:flutter_test/src/test_async_utils.dart:69:54)
I/flutter (22362): #1      WidgetTester.pump (package:flutter_test/src/widget_tester.dart:608:27)
I/flutter (22362): #2      GeneralExtendedWidgetTester.pumpUntilCondition
(file:///Users/yahya.kara/AndroidStudioProjects/dh-name-flutter/test_screens/utils/general_test_utils.dart:32:13)
I/flutter (22362): <asynchronous suspension>
I/flutter (22362): #3      GeneralExtendedWidgetTester.pumpUntilVisible
(file:///Users/yahya.kara/AndroidStudioProjects/dh-name-flutter/test_screens/utils/general_test_utils.dart:42:19)
I/flutter (22362): <asynchronous suspension>
I/flutter (22362): #4      OnboardSteps.steps.<anonymous closure>
(file:///Users/yahya.kara/AndroidStudioProjects/dh-name-flutter/integration_test/step_definitions/onboarding_steps.dart:19:15)
I/flutter (22362): <asynchronous suspension>
I/flutter (22362): <asynchronous suspension>
I/flutter (22362): #6      StepDefinitionGeneric.run.<anonymous closure> (package:gherkin/src/gherkin/steps/step_definition.dart:42:26)
I/flutter (22362): <asynchronous suspension>
I/flutter (22362): #7      Perf.measure (package:gherkin/src/utils/perf.dart:11:14)
I/flutter (22362): <asynchronous suspension>
I/flutter (22362): #8      StepDefinitionGeneric.run (package:gherkin/src/gherkin/steps/step_definition.dart:37:7)
I/flutter (22362): <asynchronous suspension>
I/flutter (22362): #9      GherkinIntegrationTestRunner.runStep (package:flutter_gherkin/src/flutter/runners/gherkin_integration_test_runner.dart:283:20)
I/flutter (22362): <asynchronous suspension>
I/flutter (22362): #10     _CustomGherkinIntegrationTestRunner.testFeature1.<anonymous closure>.<anonymous closure>
(file:///Users/yahya.kara/AndroidStudioProjects/dh-name-flutter/integration_test/gherkin_suite_test.g.dart:356:13)
I/flutter (22362): <asynchronous suspension>
I/flutter (22362): #11     GherkinIntegrationTestRunner.runScenario.<anonymous closure>
(package:flutter_gherkin/src/flutter/runners/gherkin_integration_test_runner.dart:185:13)
I/flutter (22362): <asynchronous suspension>
I/flutter (22362): #12     testWidgets.<anonymous closure>.<anonymous closure> (package:flutter_test/src/widget_tester.dart:170:15)
I/flutter (22362): <asynchronous suspension>
I/flutter (22362): #13     TestWidgetsFlutterBinding._runTestBody (package:flutter_test/src/binding.dart:806:5)
I/flutter (22362): <asynchronous suspension>
I/flutter (22362): (elided one frame from dart:async)
I/flutter (22362): 
**I/flutter (22362): When the exception was thrown, this was the stack:
I/flutter (22362): #0      TestAsyncUtils.guardSync (package:flutter_test/src/test_async_utils.dart:267:5)
I/flutter (22362): #1      TestAsyncUtils.guard (package:flutter_test/src/test_async_utils.dart:63:5)
I/flutter (22362): #2      WidgetTester.pumpAndSettle (package:flutter_test/src/widget_tester.dart:662:27)
I/flutter (22362): #3      GherkinIntegrationTestRunner.startApp (package:flutter_gherkin/src/flutter/runners/gherkin_integration_test_runner.dart:226:18)
I/flutter (22362): <asynchronous suspension>
I/flutter (22362): #4      GherkinIntegrationTestRunner.runScenario.<anonymous closure>
(package:flutter_gherkin/src/flutter/runners/gherkin_integration_test_runner.dart:165:13)
I/flutter (22362): <asynchronous suspension>
I/flutter (22362): #5      testWidgets.<anonymous closure>.<anonymous closure> (package:flutter_test/src/widget_tester.dart:170:15)
I/flutter (22362): <asynchronous suspension>
I/flutter (22362): #6      TestWidgetsFlutterBinding._runTestBody (package:flutter_test/src/binding.dart:806:5)
I/flutter (22362): <asynchronous suspension>
I/flutter (22362):** 
I/flutter (22362): The test description was:
I/flutter (22362):   Anonymous User can open order list screen with empty state
I/flutter (22362): ════════════════════════════════════════════════════════════════════════════════════════════════════
I/flutter (22362): 00:20 +0 -5: User can list added product on cart: Anonymous user should redirect to login screen while enter the basket if he added product [E]
I/flutter (22362):   Test failed. See exception logs above.
I/flutter (22362):   The test description was: Anonymous user should redirect to login screen while enter the basket if he added product
I/flutter (22362):   
I/flutter (22362): Consider enabling the flag chain-stack-traces to receive more detailed exceptions.
I/flutter (22362): For example, 'dart test --chain-stack-traces'.
I/flutter (22362): 00:20 +0 -6: (tearDownAll)
I/flutter (22362): 00:20 +0 -6: User can list added product on cart: Anonymous user should redirect to login screen while enter the basket if he added product
I/flutter (22362): Error while creating CountryRepository
I/flutter (22362): Stack trace:
I/flutter (22362):  #0      _AssertionError._doThrowNew (dart:core-patch/errors_patch.dart:47:61)
I/flutter (22362): #1      _AssertionError._throwNew (dart:core-patch/errors_patch.dart:36:5)
I/flutter (22362): #2      _GetItImplementation._findFactoryByNameAndType (package:get_it/get_it_impl.dart:372:7)
I/flutter (22362): #3      _GetItImplementation.get (package:get_it/get_it_impl.dart:393:29)
I/flutter (22362): #4      _GetItImplementation.call (package:get_it/get_it_impl.dart:430:12)
I/flutter (22362): #5      _registerRepositories.<anonymous closure> (package/core/injection_container.dart:299:83)
I/flutter (22362): #6      _ServiceFactory.getObject (package:get_it/get_it_impl.dart:160:41)
I/flutter (22362): #7      _GetItImplementation.get (package:get_it/get_it_impl.dart:409:34)
I/flutter (22362): #8      _GetItImplementation.call (package:get_it/get_it_impl.dart:430:12)
I/flutter (22362): #9      _registerUseCases.<anonymous closure> (package/core/injection_container.dart:412:89)
I/flutter (22362): #10     _ServiceFactory.getObject (package:get_it/get_it_impl.dart:154:37)
I/flutter (22362): #11     _GetItImplementation.get (package:get_it/get_it_impl.dart:409:34)
I/flutter (22362): #12     _GetItImplementation.call (pa
I/flutter (22362): Error while creating GetCountrySettingsUseCase
I/flutter (22362): Stack trace:
I/flutter (22362):  #0      _AssertionError._doThrowNew (dart:core-patch/errors_patch.dart:47:61)
I/flutter (22362): #1      _AssertionError._throwNew (dart:core-patch/errors_patch.dart:36:5)
I/flutter (22362): #2      _GetItImplementation._findFactoryByNameAndType (package:get_it/get_it_impl.dart:372:7)
I/flutter (22362): #3      _GetItImplementation.get (package:get_it/get_it_impl.dart:393:29)
I/flutter (22362): #4      _GetItImplementation.call (package:get_it/get_it_impl.dart:430:12)
I/flutter (22362): #5      _registerRepositories.<anonymous closure> (package/core/injection_container.dart:299:83)
I/flutter (22362): #6      _ServiceFactory.getObject (package:get_it/get_it_impl.dart:160:41)
I/flutter (22362): #7      _GetItImplementation.get (package:get_it/get_it_impl.dart:409:34)
I/flutter (22362): #8      _GetItImplementation.call (package:get_it/get_it_impl.dart:430:12)
I/flutter (22362): #9      _registerUseCases.<anonymous closure> (package/core/injection_container.dart:412:89)
I/flutter (22362): #10     _ServiceFactory.getObject (package:get_it/get_it_impl.dart:154:37)
I/flutter (22362): #11     _GetItImplementation.get (package:get_it/get_it_impl.dart:409:34)
I/flutter (22362): #12     _GetItImplementation.call (pa
I/flutter (22362): 00:20 +0 -6: (tearDownAll)
I/flutter (22362): 6 scenarios (6 passed)
I/flutter (22362): 9 steps (7 passed, 2 failed)
I/flutter (22362): 0:00:20.607000
I/flutter (22362): 00:20 +1 -6: Some tests failed.
D/Sentry  (22362): Envelope sent successfully.
D/Sentry  (22362): Discarding envelope from cache: /data/user/0/com.package..uat/cache/sentry/c23b7f17-7e92-49ff-880c-39feefb9cb28.envelope
D/Sentry  (22362): Envelope flushed
I/System.out(22362): (HTTPLog)-Static: isSBSettingEnabled false
I/System.out(22362): (HTTPLog)-Static: isSBSettingEnabled false```


When I check the log the tests crashes on    await tester.pumpAndSettle() of startApp method
 
 @protected
  Future<void> startApp(
    WidgetTester tester,
    World world,
  ) async {
    await appMainFunction(world);
    await tester.pumpAndSettle();
  }

**Comment by jonsamwell** [comment]
Hmmm this is suspicious?  I wonder if something when creating the app is now completing before the pumpAndSettle call?  Perhaps we don't need that pumpAndSettleCall at all?

**Comment by jonsamwell** [comment]
Should be fixed in the latest release `3.0.0-rc.10`
